### DISCUSSION TITLE
Is cheerio-advanced-selectors still compatible?

### DISCUSSION THREAD

**Discussion by literakl** [cheeriojs/cheerio#2121] [Q&A] [answered]
My code to generate sample data has started to fail with:

```
    Uncaught TypeError [ERR_INVALID_ARG_TYPE]: The "superCtor.prototype" property must be of type object. Received undefined
    NodeError             	errors.js:254
    inherits              	util.js:160
    exports.wrap          	index.js:20
    (anonymous function)  	parseAccidents.js:23
```

The line with error contains:

`    const cheerioParser = cheerioAdv.wrap(cheerio);`

I have checked out many repo revisions until I have identified the [commit](https://github.com/literakl/mezinamiridici/commit/d4e931ab90a8e5db6fb723edb3ca2ecad8636c2b) causing the troubles where I have tried in to get rid off GH security warnings and updated the NPM dependencies. Cheerio got updated but cheerio-advanced-selectors remained the same (4yrs old release).

package_lock.json

```
    "node_modules/cheerio": {
          "version": "1.0.0-rc.10" // from cheerio-1.0.0-rc.3.tgz
```

Do you have any idea how to fix the trouble?

**Comment by fb55** [comment, marked as answer]
You won't need `cheerio-advanced-selectors` anymore, as positional selectors are now natively supported by Cheerio itself.

**Comment by literakl** [comment]
Thanks, I have removed it already because I have not found any usage at
all. I will submit an issue to the library to warn the users.

ne 26. 9. 2021 v 16:01 odesílatel Felix Böhm ***@***.***>
napsal:
